src/unix/async.c

```diff
@@ -224,6 +238,14 @@ static int uv__async_start(uv_loop_t* loop) {
   if (err < 0)
     return UV__ERR(errno);
 
+  pipefd[0] = err;
+  pipefd[1] = -1;
+#elif UV__KQUEUE_EVFILT_USER
+  /* This fd will not be actually used, it's only used as a placeholder. */
+  err = uv__open_cloexec("/dev/null", O_RDONLY);
+  if (err < 0)
+    return err;
+
   pipefd[0] = err;
```

Comment:
If I understand the docs correctly, you don't need to allocate anything here, so you can instead make this an invalid fd, and also then to use this value later as a sentinel to see whether it can use EVFILT_USER and whether this needs to be closed during cleanup.
```suggestion
  /* This is not used as a fd, just an ident(ifier). */
  pipefd[0] = -2;
```

Reply:
Yes, it was supposed to be that way. I planned on using a magic number for this but something like https://github.com/libuv/libuv/blob/5d1ccc12c48099d720bb39f7430c480a52953039/src/unix/core.c#L895 and https://github.com/libuv/libuv/blob/5d1ccc12c48099d720bb39f7430c480a52953039/src/unix/core.c#L908 seemed to leave me no choice unless I'm permitted to make some code changes to those portion of the current code.

Reply:
Yeah, you don't necessarily need to go through the generic code for that, since it is only valid in kevent. Potentially similar to how libuv now deals with process exit events there also